force-app/main/default/classes/searchAndPick.cls

```diff
@@ -0,0 +1,27 @@
+global with sharing class searchAndPick {
+          private final Account person;
+          public searchAndPick(ApexPages.StandardController stdController){
+    this.person= (Account)stdController.getRecord();
+  }
+    
+     public list <Contact> ctt {get;set;}
+    public list <Contact> cttt {get;set;}
+    public list <Contact> ctct {get;set;}
+    public String searchKey {get;set;}
+     public String modKey {get;set;}
+    public searchAndPick( ) {
+    }
+    public void search(){
+        ctt = [SELECT Id, LastName, title, phone, Is_Primary_Contact__c from Contact where Accountid =:this.person.Id AND lastname like:'%'+searchKey+'%' LIMIT 10];
+}
+    public void setPrimary(){
+        	modKey = apexpages.currentpage().getparameters().get('modKey');
+			ctct = [SELECT Id, Is_Primary_Contact__c from Contact where id =:modKey AND Accountid =:this.person.Id];
+        	ctct[0].Is_Primary_Contact__c = true;
+        	update ctct;
+        	cttt = [SELECT Id, Is_Primary_Contact__c from Contact where id !=:modKey AND Accountid =:this.person.Id AND Is_Primary_Contact__c = true];
+        	cttt[0].Is_Primary_Contact__c = false;
+        	update cttt;
+        	search();
+        }
+}
```

Comment:
please add description to all methods such as

/**
    * @author Ali Ozdemir
    * @date  July 20, 2021
    * @name doRollUpsOnAccount
    * @description Calculate KPIs on Account
    * @return void
    */

Reply:
as best practice, for same type of objects, each method should have one dml operation. please put them into a list and update at once. 

please try to optimize query to avoid multiple queries. This is also a dml and method should be wrapped with try catch block.